### DISCUSSION TITLE
Using github-act-runner as a Forgejo runner

### DISCUSSION THREAD

**Discussion by earl-warren** [ChristopherHX/github-act-runner#100] [General]
Hi,

I'm evaluating the possibility of using github-act-runner as a runner for https://forgejo.org. I see that it [has support for services](https://github.com/ChristopherHX/github-act-runner/issues/96) which is an important feature. There would be a need for an adapter because the protocol is different from GitHub and I wonder if that is doable.

What do you think?

**Comment by ChristopherHX** [comment]
I'm not shure what is your exact goal.
Do you want to extend `Gitea Actions` or create a new protocol.

I'm currently using this runner with gitea with my own [actions backend](https://github.com/ChristopherHX/runner.server) (uses the same GitHub protocol to talk to the runner like GitHub and is more Actions feature complete than Gitea Actions)

> There would be a need for an adapter because the protocol is different from GitHub and I wonder if that is doable

I have already made an [adapter ](https://gitea.com/ChristopherHX/actions_runner) for `Gitea Actions` (assumes Forgejo still uses the Gitea Actions protocol) to the actions/runner protocol, but not shure if it works with `github-act-runner` only tested against [actions/runner](https://github.com/actions/runner). Maybe github-act-runner needs a small adjustment to actually work, because of some id assumptions required to workaround some gitea protocol limitations.

> I see that it [has support for services](https://github.com/ChristopherHX/github-act-runner/issues/96)

This is not designed to allow a direct integration of a different protocol, it's meant to keep compatible with the proprietary Actions Service protocol.

**Comment by earl-warren** [comment]
Thanks for the quick answer and clarification. I realize I was confused by a few things (services), my apologies. 

I also did not realize https://gitea.com/ChristopherHX/actions_runner was a runner forked from https://gitea.com/gitea/act_runner (as of commit 715d0e85ced5c95d6def01840c0f78e1eac107b6). IFAIU Its purpose is to communicate with Gitea Actions and use https://github.com/ChristopherHX/runner.server to run the steps. Which is different from what https://gitea.com/gitea/act_runner  does because it runs the steps using docker, all by itself. 

In that case https://gitea.com/ChristopherHX/actions_runner would be a better choice than https://github.com/ChristopherHX/github-act-runner to use https://github.com/ChristopherHX/runner.server.

**Reply by ChristopherHX** [reply]
> IFAIU Its purpose is to communicate with Gitea Actions and use https://github.com/ChristopherHX/runner.server to run the steps. 

Actually it's purpose is to use https://github.com/actions/runner to run the steps and let GitHub maintain the actual runner implemenation for free under the MIT license.

Yes https://gitea.com/ChristopherHX/actions_runner is a hard fork (Not forked by the fork button of Gitea) of https://gitea.com/gitea/act_runner and removed the step running code.
